src/common/pagination.ts

```diff
@@ -217,7 +217,15 @@ export class Paginator<T, U extends ListOptions> {
 
     private browseAndConcatenateAllPages(): Promise<Array<T>> {
         if (this.hasNewPage()) {
-            return this.nextPage().then( page => this.browseAndConcatenateAllPages().then( data => page.data.length === 0 ? data : page.data.concat(data)));
+            return this.nextPage()
```

Comment:
Raise bug on this

Reply:
https://jira.arm.com/browse/IOTSTOOL-2296